File: DogShelter/DogShelter.Model/ZahtjevZaUdomljavanje.cs
namespace DogShelter.Model;

public class ZahtjevZaUdomljavanje
{
    public int ZahtjevZaUdomljavanjeId { get; set; }
    public int KorisnikId { get; set; }
    public string? KorisnikIme { get; set; }
    public string? KorisnikPrezime { get; set; }
    public int PasId { get; set; }
    public string? PasNaziv { get; set; }
    public string? PasSlikaNaslovna { get; set; }

    // Lets the client disable "Odobri" (rather than let the click round-trip to Odobri()'s own
    // Aktivan/Dostupan re-check) when the dog stopped being adoptable after the request was
    // submitted - e.g. deactivated, or already reserved/adopted via a different request.
    public bool PasAktivan { get; set; }
    public string? PasStatusNaziv { get; set; }
    public int StatusZahtjevaId { get; set; }
    public string? StatusZahtjevaNaziv { get; set; }
    public DateTime DatumPodnosenja { get; set; }
    public string? Napomena { get; set; }
    public DateTime? DatumObrade { get; set; }
    public int? ObradioKorisnikId { get; set; }
    public string? ObradioKorisnikIme { get; set; }
    public string? ObradioKorisnikPrezime { get; set; }
    public string? RazlogOdbijanja { get; set; }

    // True once FinalizirajUdomljenje has actually run for this request (an Udomljavanje row
    // exists) - lets clients tell an Odobren-but-not-yet-finalized request (dog Rezervisan, action
    // still pending) apart from an Odobren request that was already finalized long ago (dog
    // Udomljen), since StatusZahtjevaNaziv stays "Odobren" forever in both cases.
    public bool UdomljenjeFinalizovano { get; set; }
}

File: dogshelter_mobile/lib/features/adoption_requests/domain/zahtjev_za_udomljavanje.dart
class ZahtjevZaUdomljavanje {
  ZahtjevZaUdomljavanje({
    required this.zahtjevZaUdomljavanjeId,
    required this.korisnikId,
    this.korisnikIme,
    this.korisnikPrezime,
    required this.pasId,
    this.pasNaziv,
    this.pasSlikaNaslovna,
    required this.statusZahtjevaId,
    this.statusZahtjevaNaziv,
    required this.datumPodnosenja,
    this.napomena,
    this.datumObrade,
    this.obradioKorisnikId,
    this.obradioKorisnikIme,
    this.obradioKorisnikPrezime,
    this.razlogOdbijanja,
  });

  final int zahtjevZaUdomljavanjeId;
  final int korisnikId;
  final String? korisnikIme;
  final String? korisnikPrezime;
  final int pasId;
  final String? pasNaziv;
  final String? pasSlikaNaslovna;
  final int statusZahtjevaId;
  final String? statusZahtjevaNaziv;
  final DateTime datumPodnosenja;
  final String? napomena;
  final DateTime? datumObrade;
  final int? obradioKorisnikId;
  final String? obradioKorisnikIme;
  final String? obradioKorisnikPrezime;
  final String? razlogOdbijanja;

  factory ZahtjevZaUdomljavanje.fromJson(Map<String, dynamic> json) {
    return ZahtjevZaUdomljavanje(
      zahtjevZaUdomljavanjeId: json['zahtjevZaUdomljavanjeId'] as int,
      korisnikId: json['korisnikId'] as int,
      korisnikIme: json['korisnikIme'] as String?,
      korisnikPrezime: json['korisnikPrezime'] as String?,
      pasId: json['pasId'] as int,
      pasNaziv: json['pasNaziv'] as String?,
      pasSlikaNaslovna: json['pasSlikaNaslovna'] as String?,
      statusZahtjevaId: json['statusZahtjevaId'] as int,
      statusZahtjevaNaziv: json['statusZahtjevaNaziv'] as String?,
      datumPodnosenja: DateTime.parse(json['datumPodnosenja'] as String),
      napomena: json['napomena'] as String?,
      datumObrade: json['datumObrade'] == null ? null : DateTime.parse(json['datumObrade'] as String),
      obradioKorisnikId: json['obradioKorisnikId'] as int?,
      obradioKorisnikIme: json['obradioKorisnikIme'] as String?,
      obradioKorisnikPrezime: json['obradioKorisnikPrezime'] as String?,
      razlogOdbijanja: json['razlogOdbijanja'] as String?,
    );
  }
}
